src/dmd/typesem.d

```diff
@@ -4053,13 +4053,36 @@ Expression dotExp(Type mt, Scope* sc, Expression e, Identifier ident, int flag)
                 /* Rewrite as:
                  *  this.d
                  */
-                if (hasThis(sc))
+                AggregateDeclaration ad = d.isMemberLocal();
+                if (auto f = hasThis(sc))
                 {
                     // This is almost same as getRightThis() in expression.c
```

Comment:
Does this comment still apply? (Also please update the file mentioned).

Reply:
It still does. updated.

Reply:
Thanks.